internal/commands/user/inputs.go

```diff
@@ -47,3 +57,56 @@ func displayUserData(apt realm.AuthProviderType, user realm.User) string {
 	}
 	return ""
 }
+
+func (i multiUserInputs) filter() realm.UserFilter {
+	return realm.UserFilter{
+		IDs:       i.Users,
+		State:     i.State,
+		Pending:   i.Pending,
+		Providers: realm.NewAuthProviderTypes(i.ProviderTypes...),
+	}
+}
+
+func (i multiUserInputs) resolveUsers(realmClient realm.Client, groupID, appID string) ([]realm.User, error) {
+	foundUsers, findErr := realmClient.FindUsers(groupID, appID, i.filter())
+	if findErr != nil {
+		return nil, findErr
+	}
+	if len(i.Users) > 0 && len(foundUsers) == 0 {
+		return nil, errors.New("no users found")
+	}
+	return foundUsers, nil
+}
+
+func (i multiUserInputs) selectUsers(ui terminal.UI, resolvedUsers []realm.User, action string) ([]realm.User, error) {
+	if len(i.Users) > 0 {
```

Comment:
looking at this now, we could probably turn this if check into `if len(resolvedUsers) == 0` and just return `nil, nil` if thats the case.  in that case, we don't need this function to be defined on `multiUserInputs` either, it could just be:

```go
func selectUsers(ui terminal.UI, users []realm.User, action string) ([]realm.User, error) {
  if len(users) == 0 { return nil, nil }
  
  selected := make([]realm.User, len(selections))
  // do rest of select logic here...
  return select, nil
}
```

Reply:
The issue came across here was a scenario where the user flag is passed in, and those users are found before this function calls. If we don't user the multiUserInputs.Users field, we will prompt the user for selection even though that step isn't necessary. I updated the logic to check both the multiUserInputs.Users and users param so we can account for all scenarios

Reply:
Discussed offline, leaving as is